.envrc

```diff
@@ -0,0 +1,153 @@
+# This is the .envrc for Sentry, for use with direnv.
+# We use it for:
+#   - lightweight (crucial since it's run on every cd to here) state checking and assurance (e.g. activating the venv)
+#   - one-time heavyweight state initialization (such as creating a .venv)
+#       - TODO: heavyweight state assurance, such as syncing venv, should happen in a one-off "fixall" script
+#               and should be recommended by direnv as a catch-all fixer
+#       - Although, this kind of state initialization could be moved to there from here in the future.
+#   - setting useful environment variables
+
+export PYTHONDONTWRITEBYTECODE=1
+export PIP_DISABLE_PIP_VERSION_CHECK=on
+export NODE_OPTIONS=--max-old-space-size=4096
+
+bold="$(tput bold)"
+red="$(tput setaf 1)"
+reset="$(tput sgr0)"
+
+# NOTE: do not trap bash's EXIT; it will prevent direnv from performing its final dance(s)
+trap notice ERR
+
+notice () {
+    [ $? -eq 0 ] && return
+    cat <<EOF
+${bold}direnv tooling is in an ALPHA state!
+If you're having trouble, or have questions, please ask in #discuss-dev-tooling
+and/or reach out to @josh.
+${reset}
+EOF
+}
+
+require () {
+    command -v "$1" 2>&1 > /dev/null
+}
+
+info () {
+    cat <<EOF
+${bold}direnv: ${1}
+${reset}
+EOF
+}
+
+die () {
+    >&2 cat <<EOF
+${red}${bold}direnv FATAL: ${1}
+${reset}
+EOF
+    exit 1
+}
+
+init_venv () {
+    deactivate 2>/dev/null || true
+    python2.7 -c "import sys; sys.exit(sys.version_info[:2] != (2, 7))" || \
```

Comment:
Why would you even need this if you're explicitly calling `python2.7` binary? I mean, I agree that that doesn't guarantee it, but you'd have to go out of your way to do something really odd here like symlink python3 to python2.7 or something.

Reply:
Good point. Admittedly this was copypaste from one of my old branches where I didn't have a following `python2.7 -m virtualenv .venv`. I'll replace it with a command check.